pkg/recommender.py

```diff
@@ -145,6 +156,20 @@ def dispatch(self, speaker_id, evt):
         if not isinstance(evt, np.ndarray):
             evt = np.array(evt)
 
+        #safety in case of numpy array or float
+        if type(speaker_id) is not int:
+            raise TypeError('Speaker id must be integer, received: ' + speaker_id)
+
+        # system must be initialized
+        if not self.sched_initialized:
+            log('Scheduled events not yet initialized')
+            return
+
+        #acoustic events only sent during time interval or not during current scheduled events
+        if not self.recomm_start:
+            log('Current time outside acceptable time interval or scheduled events in progress')
+            return
+
         thread = Thread(target=self._process_evt, args=(speaker_id, evt))
         thread.daemon = True
         thread.start()
```

Comment:
Hey! Is there a good way to kill threads? Currently, we have implemented logic that does not allow recommendation messages to interrupt scheduled events. Meaning a new recommendation thread can not be initialized if scheduled events are in progress. However, the opposite is not completely true. If there is a recommendation thread currently in progress and it's time for the evening messages, sometimes weird things happen. I have tried to do my best to alleviate this issue by adding some conditional statements in the `call_poll_ema` function, but it is still not perfect. Is there a way to kill the recommendation thread when the evening messages are about to start?